recipe-client-addon/test/browser/browser_NormandyDriver.js

```diff
@@ -12,6 +12,48 @@ add_task(withDriver(Assert, async function uuids(driver) {
   isnot(uuid1, uuid2, "uuids are unique");
 }));
 
+add_task(withDriver(Assert, async function installXpi(driver) {
+  // Test that we can install an XPI from any URL
+  const dir = getChromeDir(getResolvedURI(gTestPath));
+  dir.append("fixtures");
+  dir.append("normandy.xpi");
+  const xpiUrl = Services.io.newFileURI(dir).spec;
+  var addonId = await driver.installAddon(xpiUrl);
+  is(addonId, "normandydriver@example.com", "Expected test addon was installed");
+  isnot(addonId, null, "Addon install was successful");
+
+  // Installing kicks off an asychronous process which tries to read the manifest.json
+  // to startup the addon. Note that onInstallEnded is triggered too early
+  // which is why we need this delay.
+  await new Promise(resolve => SimpleTest.executeSoon(resolve));
```

Comment:
More generally, it feels odd that the addon manager doesn't provide something 'better' to wait for, where install is 'properly' finished, instead of only sort of... Maybe @Mossop  can suggest something?